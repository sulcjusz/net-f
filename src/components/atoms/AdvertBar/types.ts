export interface IAdvertBarProps {
  children: React.ReactNode
  className?: string
}
