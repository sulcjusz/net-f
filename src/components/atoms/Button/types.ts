export interface IButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  disabled?: boolean
  children: React.ReactNode
  className?: string
}
