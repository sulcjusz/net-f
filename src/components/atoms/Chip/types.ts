export interface IChipProps {
  size?: 'small' | 'medium'
  textColor?: 'black' | 'white'
  children: React.ReactNode
  className?: string
}
