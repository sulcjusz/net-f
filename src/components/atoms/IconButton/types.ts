export interface IIconButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  disabled?: boolean
  icon: React.ReactNode
  className?: string
}
