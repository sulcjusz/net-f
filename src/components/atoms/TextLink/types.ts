export interface ITextLinkProps {
  text: string
  onClick: () => void
  className?: string
}
