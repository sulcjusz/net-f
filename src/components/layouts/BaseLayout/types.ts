export interface IBaseLayoutProps {
  children: React.ReactNode
}
