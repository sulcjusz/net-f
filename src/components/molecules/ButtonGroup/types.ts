export interface IButtonGroupProps {
  children: React.ReactNode
}
