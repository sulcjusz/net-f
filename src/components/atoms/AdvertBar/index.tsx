import React from 'react'
import classNames from 'classnames'
import {IAdvertBarProps} from './types'

const AdvertBar: React.FC<IAdvertBarProps> = ({children, className}) => (
  <>
    <div
      className={classNames(
        'flex items-center justify-center px-6 py-4 bg-grey-300 ',
        className
      )}
    >
      {children}
    </div>
    <div className="min-h-[9px] bg-linear-gradient-primary"></div>
  </>
)

export default AdvertBar
