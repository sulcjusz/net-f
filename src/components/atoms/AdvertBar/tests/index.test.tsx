import {render, RenderResult} from '@testing-library/react'
import '@testing-library/jest-dom'

import AdvertBar from '../index'
import {IAdvertBarProps} from '../types'

type AdvertBarRenderProps = Partial<IAdvertBarProps>

const renderAdvertBar = (
  props?: AdvertBarRenderProps
): {advertBar: HTMLElement} & RenderResult => {
  const utils = render(<AdvertBar {...props}>Example AdvertBar</AdvertBar>)
  const advertBar = utils.getByText('Example AdvertBar')
  return {...utils, advertBar}
}

describe('AdvertBar component', () => {
  it('should match the snapshot', () => {
    const {advertBar} = renderAdvertBar()
    expect(advertBar.firstChild).toMatchSnapshot()
  })

  it('renders correctly with default props', () => {
    const {advertBar} = renderAdvertBar()
    expect(advertBar).toBeInTheDocument()
    expect(advertBar).toHaveClass(
      'flex items-center justify-center px-6 py-4 bg-grey-300'
    )
    expect(advertBar).not.toHaveClass('bg-linear-gradient-primary')
  })

  it('renders correctly with custom className', () => {
    const {advertBar} = renderAdvertBar({className: 'custom-class'})
    expect(advertBar).toHaveClass('custom-class')
  })

  it('renders children correctly', () => {
    const {advertBar} = renderAdvertBar()
    expect(advertBar.textContent).toBe('Example AdvertBar')
  })
})
