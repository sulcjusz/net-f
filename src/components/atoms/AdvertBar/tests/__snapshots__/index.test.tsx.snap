// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvertBar component should match the snapshot 1`] = `Example AdvertBar`;
